va/caa.go

```diff
@@ -173,63 +178,68 @@ func (va *ValidationAuthorityImpl) getCAASet(ctx context.Context, hostname strin
 	//
 	// We depend on our resolver to snap CNAME and DNAME records.
 	results := va.parallelCAALookup(ctx, hostname)
-	return parseResults(results)
+	return selectCAA(results)
 }
 
 // checkCAARecords fetches the CAA records for the given identifier and then
 // validates them. If the identifier argument's value has a wildcard prefix then
 // the prefix is stripped and validation will be performed against the base
 // domain, honouring any issueWild CAA records encountered as appropriate.
-// checkCAARecords returns four values: the first is a bool indicating whether
-// CAA records were present after filtering for known/supported CAA tags. The
+// checkCAARecords returns four values: the first is a string indicating at
+// which name (i.e. FQDN or parent thereof) CAA records were found, if any. The
 // second is a bool indicating whether issuance for the identifier is valid. The
 // unmodified *dns.CAA records that were processed/filtered are returned as the
 // third argument. Any  errors encountered are returned as the fourth return
 // value (or nil).
 func (va *ValidationAuthorityImpl) checkCAARecords(
 	ctx context.Context,
 	identifier identifier.ACMEIdentifier,
-	params *caaParams) (bool, bool, string, error) {
+	params *caaParams) (string, bool, string, error) {
 	hostname := strings.ToLower(identifier.Value)
 	// If this is a wildcard name, remove the prefix
 	var wildcard bool
 	if strings.HasPrefix(hostname, `*.`) {
 		hostname = strings.TrimPrefix(identifier.Value, `*.`)
 		wildcard = true
 	}
-	caaSet, response, err := va.getCAASet(ctx, hostname)
+	caaSet, err := va.getCAA(ctx, hostname)
 	if err != nil {
-		return false, false, "", err
+		return "", false, "", err
+	}
+	raw := ""
```

Comment:
This is a total nitpick, but `raw` will take the zero-value of `""` rather than it being explicitly assigned by the short declaration.
```suggestion
	var raw string
```

Reply:
The go compiler will optimize this away under the hood. It's easier for a human reading through the code to not have to do the mental gymnastics to realize that the zero-value of a string is `""`.